include/nbl/builtin/hlsl/ieee754.hlsl

```diff
@@ -0,0 +1,128 @@
+#ifndef _NBL_BUILTIN_HLSL_IEE754_H_INCLUDED_
+#define _NBL_BUILTIN_HLSL_IEE754_H_INCLUDED_
+
+#include <nbl/builtin/hlsl/type_traits.hlsl>
+
+namespace nbl::hlsl::ieee754
+{
+	namespace impl
+	{
+		template <typename T>
+		unsigned_integer_of_size<sizeof(T)>::type castToUintType(T x)
+		{
+			using AsUint = typename unsigned_integer_of_size<sizeof(T)>::type;
+			return bit_cast<AsUint, T>(x);
+		}
+
+		// to avoid bit cast from uintN_t to uintN_t
+		template <> unsigned_integer_of_size<2>::type castToUintType(uint16_t x) { return x; }
+		template <> unsigned_integer_of_size<4>::type castToUintType(uint32_t x) { return x; }
+		template <> unsigned_integer_of_size<8>::type castToUintType(uint64_t x) { return x; }
+
+		template <typename T>
+		T replaceBiasedExponent(T x, uint32_t biasedExp)
```

Comment:
IMHO the `biasedExp` needs to be promoted to a size that matches T cause you might risk truncation when shifting its value up